### ISSUE TITLE
classes.ts:22 Uncaught ReferenceError: process is not defined

### ISSUE THREAD

**Issue by JenuelDev** [palantir/blueprint#4829]
#### Environment

- __Package version(s)__: "^3.47.0"
- __Operating System__: windows 10
- __Browser name and version__: Chrome 92.+

#### Code Sandbox
[link to sandbox](https://codesandbox.io/s/romantic-moser-3zhx8?file=/src/App.tsx)

#### Steps to reproduce

1. git clone https://github.com/BroJenuel-Youtube/testing-vite-react
2. yarn dev
3. Button not working!

#### Actual behavior

I tried using ***VITE*** for my react project, used a button but not working. and this error shows up in my console.
![image](https://user-images.githubusercontent.com/31676163/127736430-18b6f3fa-0921-4c28-96d9-2da09a43c872.png)

here is the code
```jsx
import React, { useState } from 'react'
import logo from './logo.svg'
import './App.css'
import { Button } from "@blueprintjs/core";

function App() {
  const [count, setCount] = useState(0)

  return (
    <div className="App">
      <Button intent="success" text="button content" onClick={() => setCount((count) => count +1)} />
      Hello World! {count}
      <button onClick={() => setCount((count) => count +1)} >
        Button
      </button>
    </div>
  )
}

export default App

```

#### Expected behavior
The button component should run.

**Comment by adidahiya** [comment]
I'm not familiar with Vite. This looks like a duplicate of https://github.com/palantir/blueprint/issues/3739. Until that issue is fixed, maybe you can define `process.env` through Vite config?

**Comment by adidahiya** [comment]
fixed by #4934
